### ISSUE TITLE
Modernize or remove step-17.

### ISSUE THREAD

**Issue by drwells** [dealii/dealii#15648]
@labdala and I looked at this a few weeks ago and there are a bunch of things that need to be modernized in this program.

Before we do this, lets review what step-40 does and make sure that we still need this tutorial. If 40 covers everything we need then we shouldn't have both (and we should delete this tutorial program). That being said:

- [ ] Move some of the discussion on `IndexSet`s to step-17 from step-40 (we may want to make this depend on step-40 instead; worth thinking about)
- [ ] Switch to `parallel::shared::Triangulation` (using artificial cells)
- [ ] Avoid `std::vector<IndexSet>`
- [ ] Have a good discussion on parallelization
- [ ] Don't use `MatrixTools`
- [ ] Don't loop over quadrature points in the innermost loop
- [ ] Use a better weak form

**Comment by drwells** [comment]
We just noticed that step-18 does `p::s::T` so this tutorial can probably be deleted.

**Comment by bangerth** [comment]
step-17 and -18 are wonky for many reasons. step-18 uses a formulation of large-deformation elasticity that nobody else uses (and that may be unphysical), but it introduces `p::s::T` and `QuadraturePointData`, and so it has value. I don't know off the top of my head what step-17 provides other than being the first program that uses PETSc.

**Comment by labdala** [comment]
Yes, step 17's appeal to me was the first-time take on PETSc wrapper classes. As a beginner, it felt more tangible than reading step-40 straight.
